### PR TITLE
Remove: macOS support before 10.7

### PR DESCRIPTION
**Pull request by michicc**
Updated and somewhat adapted from PR #8025, split a bit more into feature sets in case somebody has a need to revert a specific area. Target goal was 10.7 because that is the first version with any C++11 support, and almost nothing is gained by targeting 10.9.

Since the C++11 move, getting OpenTTD to compile and run for anything below 10.7 basically requires building a custom compiler and libc++. As such, this PR removes any special support for compiling with SDK versions before 10.7 and chucks most of the alternative, compatibility API support for video/sound/music/text.

The remaining implementations use whatever API level is still current, which is mostly something between 10.5 and 10.7. It should still be possible to compile OpenTTD targeting 10.7 with this PR.

I left some compatibility checks in that don't create any maintenance burdens or alternative code paths.

### CODE REVIEW

File: src/video/cocoa/cocoa_v.mm [hunk 1]
```diff
@@ -395,35 +395,11 @@ void QZ_GameSizeChanged()
  */
 static CocoaSubdriver *QZ_CreateWindowSubdriver(int width, int height, int bpp)
 {
-#if defined(ENABLE_COCOA_QUARTZ) || defined(ENABLE_COCOA_QUICKDRAW)
-	CocoaSubdriver *ret;
-#endif
-
 #if defined(ENABLE_COCOA_QUARTZ) && (MAC_OS_X_VERSION_MAX_ALLOWED >= MAC_OS_X_VERSION_10_4)
```
Comment (nielsmh) [inline src/video/cocoa/cocoa_v.mm]: Can't this one be simplified away too? ENABLE_COCOA_QUARTZ should always be defined anyway.

Comment (michicc) [reply src/video/cocoa/cocoa_v.mm]: Somebody might in theory want to build a headless dedicated server.

File: src/blitter/factory.hpp [hunk 2]
```diff
@@ -114,17 +111,7 @@ class BlitterFactory {
...
-#if defined(WITH_COCOA)
-		/* Some people reported lack of fullscreen support in 8 bpp mode.
-		 * While we prefer 8 bpp since it's faster, we will still have to test for support. */
-		if (!QZ_CanDisplay8bpp()) {
-			/* The main display can't go to 8 bpp fullscreen mode.
-			 * We will have to switch to 32 bpp by default. */
-			default_blitter = "32bpp-anim";
-		}
-#endif /* defined(WITH_COCOA) */
+		const char *default_blitter = "32bpp-anim";
```
Comment (frosch123) [inline src/blitter/factory.hpp:114]: This is a change for non-OSX.

Comment (michicc) [reply src/blitter/factory.hpp:114]: Eh, deleted too much I guess...

**Review by frosch123 — APPROVED** [review]

**Pull request merged**